src/main/java/org/vanilladb/core/util/TransactionProfiler.java

```diff
@@ -0,0 +1,282 @@
+import java.util.List;
+import java.util.Map;
+
+public class TransactionProfiler {
+
+	private static final String TOTAL_KEY = "Total";
+	private static final boolean ENABLE_CPU_TIMER = true;
+	private static final boolean ENABLE_DISKIO_COUNTER = true;
+
+	private static final ThreadLocal<TransactionProfiler> LOCAL_PROFILER = new ThreadLocal<TransactionProfiler>() {
+		@Override
+		protected TransactionProfiler initialValue() {
+			return new TransactionProfiler();
+		}
+	};
+
+	/**
+	 * Get the profiler local to this thread.
+	 * 
+	 * @return the local profiler
+	 */
+	public static TransactionProfiler getLocalProfiler() {
+		return LOCAL_PROFILER.get();
+	}
+
+	private static class SubProfiler {
+		private int count = 0, startTimes = 0;
+		private long timeStart = 0, totalTime = 0;
+		private long cpuStart = 0, totalCpuTime = 0;
+		private long ioStart = 0, totalIOCount = 0;
+		
+		public void setStartProfile(long start) {
+			if (startTimes == 0) 
+				this.timeStart = start;
+			startTimes++;
+			count++;
+		}
+		
+		public void setStartProfile(long start, long cpuStart, int ioStart) {
+			if (startTimes == 0) {
+				this.timeStart = start;
+				this.cpuStart = cpuStart;
+				if (ENABLE_DISKIO_COUNTER)
+					this.ioStart = ioStart;
+			}
+			startTimes++;
+			count++;
+		}
+		
+		public void setStopProfile(long stop) {
+			startTimes--;
+			if (startTimes == 0)
+				totalTime += (stop - cpuStart) / 1000;
+		}
+		
+		public void setStopProfile(long stop, long cpuStop, int ioStop) {
+			startTimes--;
+			if (startTimes == 0) {
+				totalTime += (stop - timeStart) / 1000;
+				totalCpuTime += (cpuStop - cpuStart) / 1000;
+				if (ENABLE_DISKIO_COUNTER)
+					totalIOCount += ioStop - ioStart;
+			}	
+		}
+		
+		public void startProfiler(int ioStart) {
+			if (startTimes == 0)
+				timeStart = System.nanoTime();
+			if (ENABLE_CPU_TIMER)
+				cpuStart = ThreadMXBean.getCpuTime();
+			if (ENABLE_DISKIO_COUNTER)
+				this.ioStart = ioStart;	
+			startTimes++;
+			count++;
+		}
+
+		public void stopProfiler(int ioStop) {
+			startTimes--;
+			if (startTimes == 0) {
+				totalTime += (System.nanoTime() - timeStart) / 1000;
+				if (ENABLE_CPU_TIMER)
+					totalCpuTime = (ThreadMXBean.getCpuTime()-cpuStart) / 1000;
+				if (ENABLE_DISKIO_COUNTER)
+					this.totalIOCount = ioStop - ioStart;
+			}
+		}
+
+		public long getTotalTime() {
+			return totalTime;
+		}
+		
+		public long getTotalCpuTime() {
+			return totalCpuTime;
+		}
+		
+		public long getTotalIOCount() {
+			return totalIOCount;
+		}
+
+		public long getCount() {
+			return count;
+		}
+	}
+
+	private Map<Object, SubProfiler> subProfilers = new HashMap<Object, SubProfiler>();
+	// We want to preserve the order of creating profilers so that
+	// we use a list to record the order.
+	private List<Object> componenents = new LinkedList<Object>();	
+	private int ioCount = 0;
+
+	public void reset() {
+		subProfilers.clear();
+		componenents.clear();
+	}
+
+	public void increment() {

```

Comment:
It would be better to use `incrementIoCount`, or it might be hard for users to know what is incremented.